src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3525,19 +3542,25 @@ void ApplicationManagerImpl::ProcessApp(const uint32_t app_id,
 void ApplicationManagerImpl::ClearTimerPool() {
   LOG4CXX_AUTO_TRACE(logger_);
 
-  std::vector<TimerSPtr> new_timer_pool;
-
-  sync_primitives::AutoLock lock(timer_pool_lock_);
-  new_timer_pool.push_back(timer_pool_[0]);
-
-  for (size_t i = 1; i < timer_pool_.size(); ++i) {
-    if (timer_pool_[i]->is_running()) {
-      new_timer_pool.push_back(timer_pool_[i]);
+  auto clear_timer_pool = [](std::vector<TimerSPtr> vector_to_clear) {
```

Comment:
@mkorniichuk whole function can be replaced with `copy_if` + `clear`

Reply:
@AKalinich-Luxoft fixed in 3d068569c1b872894e5596413a51bddc49747ab0